e2e/vsphere_test.go

```diff
@@ -0,0 +1,225 @@
+package e2e
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	vspherev1 "sigs.k8s.io/cluster-api-provider-vsphere/apis/v1beta1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	yaml "sigs.k8s.io/yaml"
+
+	configv1 "github.com/openshift/api/config/v1"
+	mapiv1 "github.com/openshift/api/machine/v1beta1"
+	"github.com/openshift/cluster-capi-operator/e2e/framework"
+)
+
+const (
+	vSphereMachineTemplateName = "vsphere-machine-template"
+	kubeSystemnamespace        = "kube-system"
+	vSphereCredentialsName     = "vsphere-creds"
+)
+
+var _ = Describe("Cluster API vSphere MachineSet", Ordered, func() {
+	var vSphereMachineTemplate *vspherev1.VSphereMachineTemplate
+	var machineSet *clusterv1.MachineSet
+	var mapiMachineSpec *mapiv1.VSphereMachineProviderSpec
+
+	BeforeAll(func() {
+		if platform != configv1.VSpherePlatformType {
+			Skip("Skipping vSphere E2E tests")
+		}
+		mapiMachineSpec = getVSphereMAPIProviderSpec(cl)
+		createVSphereSecret(cl, mapiMachineSpec)
+		framework.CreateCoreCluster(cl, clusterName, "VSphereCluster")
+		createVSphereCluster(cl, mapiMachineSpec)
+	})
+
+	AfterEach(func() {
+		if platform != configv1.VSpherePlatformType {
+			// Because AfterEach always runs, even when tests are skipped, we have to
+			// explicitly skip it here for other platforms.
```

Comment:
Alternative is to use `DeferCleanup`, but I think this works

Reply:
Yeah, this is a copy pasta from other platforms _test files.
We might want to revisit this for all of them when we review testing, I'll keep it like this for now to avoid differences.